internal/nodes/avi_model_l4_translator.go

```diff
@@ -385,3 +388,51 @@ func GetDefaultSubDomain() []string {
 	}
 	return cloudProperty.NSIpamDNS
 }
+
+func buildL4InfraSetting(vs *AviVsNode, vsvip *AviVSVIPNode, svc *corev1.Service, advl4GWClassName *string) {
+	var err error
+	var infraSetting *akov1alpha1.AviInfraSetting
+
+	if lib.UseServicesAPI() && advl4GWClassName != nil {
+		gwClass, err := lib.GetSvcAPIInformers().GatewayClassInformer.Lister().Get(*advl4GWClassName)
+		if err != nil {
+			utils.AviLog.Warnf("Unable to get corresponding GatewayClass %s", err.Error())
+		} else {
+			if gwClass.Spec.ParametersRef != nil && gwClass.Spec.ParametersRef.Group == lib.AkoGroup && gwClass.Spec.ParametersRef.Kind == lib.AviInfraSetting {
+				infraSetting, err = lib.GetCRDInformers().AviInfraSettingInformer.Lister().Get(gwClass.Spec.ParametersRef.Name)
+				if err != nil {
+					utils.AviLog.Warnf("Unable to get corresponding AviInfraSetting: %s", err.Error())
+				}
+			}
+		}
+	} else {
+		if infraSettingAnnotation, ok := svc.GetAnnotations()[lib.InfraSettingNameAnnotation]; ok && infraSettingAnnotation != "" {
+			infraSetting, err = lib.GetCRDInformers().AviInfraSettingInformer.Lister().Get(infraSettingAnnotation)
+			if err != nil {
+				utils.AviLog.Warnf("Unable to get corresponding AviInfraSetting: %s", err.Error())
+			}
+		}
+	}
+
+	applyInfraSetting := false
+	if infraSetting != nil && err == nil && infraSetting.Status.Status == lib.StatusAccepted {
```

Comment:
I think this function buildL4InfraSetting can be re organised a bit.
If vs.ServiceEngineGroup == "" { } block can be put at towards the beginning  of the function so that it is applied for all. 
In case of error, return. 
then 
if infraSetting != nil && infraSetting.Status.Status == lib.StatusAccepted {
// do tasks for applyInfraSetting
}

probably we don't need the variable applyInfraSetting